clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fmap.py

```diff
@@ -245,165 +253,217 @@ def fmap_image(
-                if os.path.exists(nifti_file):
-                    phase_nifti = os.path.join(
-                        subroot, filename[:-12] + "phase1.nii.gz"
-                    )
-                    os.rename(nifti_file, phase_nifti)
-            if filename[:-15] == "fmap_e2_ph.json":
-                phase_json = os.path.join(subroot, filename[:-15] + "phase2.json")
-                os.rename(fmap_filename, phase_json)
-                if os.path.exists(nifti_file):
-                    phase_nifti = os.path.join(
-                        subroot, filename[:-15] + "phase2.nii.gz"
-                    )
-                    os.rename(nifti_file, phase_nifti)
+            raise NotImplementedError(
+                f"No renaming should be performed for case {case.value}"
+            )
+        return "phasediff"
+    raise ValueError(f"Extension {old_extension} not taken in charge.")
+
+
+def rename_files(fmap_path: Path, filenames: Iterable[str], case: BIDSFMAPCase):
+    pattern = r"(.*_)fmap_(.*?)(\..*)"
+    for previous_filename in filenames:
+        rgx = re.search(pattern, previous_filename)
+        cut, extension, type = rgx.group(1), rgx.group(2), rgx.group(3)
+        new_name = f"{cut}{phase_magnitude_renamer(extension, case)}{type}"
+        os.rename(fmap_path / previous_filename, fmap_path / new_name)
+
+
+def fmap_case_handler_factory(case: BIDSFMAPCase) -> Callable:
+    if case == BIDSFMAPCase.EMPTY_FOLDER:
+        return empty_folder_handler
+    if case == BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES:
+        return one_phase_two_magnitudes_handler
+    if case == BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES:
+        return two_phases_two_magnitudes_handler
+    if case == BIDSFMAPCase.DIRECT_FIELDMAPS:
+        return direct_fieldmaps_handler
+    if case == BIDSFMAPCase.NOT_SUPPORTED:
+        return not_supported_handler
+
+
+def empty_folder_handler(fmap_path: Path):
+    sub = fmap_path.parent.parent.name
+    ses = fmap_path.parent.name
+    cprint(f"Folder for {sub}, {ses} is empty", lvl="warning")
+
+
+def one_phase_two_magnitudes_handler(fmap_path: Path):
+    "Performs the checks and renaming associated to BIDS specifications, case 1 for fieldmaps"
+    sub = fmap_path.parent.parent.name
+    ses = fmap_path.parent.name
+
+    cprint(
+        f"BIDS Case 1 : expecting 1 phasediff and at least 1 magnitude files for {sub}, {ses}.",
+        lvl="info",
+    )
+
+    files = [f for f in os.listdir(fmap_path) if not f.startswith(".")]
+    js = [f for f in files if f.endswith("ph.json")][0]
+
+    # Checking for Echotime keys in phase json
+    with open(fmap_path / js, "r") as file:
+        json_data = json.load(file)
+    if "EchoTime1" not in json_data or "EchoTime2" not in json_data:
+        cprint(f'Invalid file {js}, missing "EchoTime" key.', lvl="warning")
+        unrecognized_fmap_case(fmap_path)
+        return
+    # Renaming
+    rename_files(fmap_path, files, BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES)
```

Comment:
Now that I see this, I think we can simplify this API by avoiding passing `files` as it can be easily computed from `fmap_path` directly in `rename_files`. I think we can afford the performance cost.